instance/tests/integration/integration_instance.py

```diff
@@ -79,6 +81,41 @@ def assert_swift_container_provisioned(self, instance):
         stat_result = stat_container(instance.swift_container_name)
         self.assertEqual(stat_result.read_acl, '.r:*')
 
+    def assert_mysql_db_provisioned(self, instance):
+        """
+        Verify that the MySQL database for the instance has been provisioned and can be
+        connected to with the credentials the instance provides.
+        """
+        for database in instance.mysql_databases:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=database['user'],
+                passwd=instance._get_mysql_pass(database['user']),
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            database_name = connection.escape_string(database['name']).decode()
+            cur = connection.cursor()
+            result = cur.execute(
+                "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '{db_name}'".format(
+                    db_name=database_name
+                )
```

Comment:
Instead of using `connection.escape_string` together with `format` (any use of `format` to pass data via SQL should be suspect), can't you just do this more simple/secure version?

```
result = cur.execute("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = %s", (db_name, ))
```